bugsnag-android-core/src/main/java/com/bugsnag/android/Metadata.kt

```diff
@@ -80,7 +79,17 @@ internal class Metadata @JvmOverloads constructor(map: Map<String, Any> = Concur
         }
     }
 
-    fun toMap(): Map<String, Any> = HashMap(store)
+    fun toMap(): Map<String, Any> {
+        val hashMap = HashMap(store)
+
+        // deep copy each section
+        store.entries.forEach {
+            if (it.value is ConcurrentHashMap<*, *>) {
+                hashMap[it.key] = ConcurrentHashMap(it.value as ConcurrentHashMap<*, *>)
```

Comment:
Do we not have to clone nested complex objects too?

Reply:
No - these would have been added by the user as a metadata value and it would be confusing to clone this IMO.

We would also not know how to clone nested objects unless they were a Map/other known type.